File: r/sparklabs/batch/sales/sql/dq_data_quality.R
source("r/sparklabs/batch/sales/sql/utils.R")

duplicates <- run_sql("
  SELECT customer_id, COUNT(*) AS duplicate_count
  FROM customers
  GROUP BY customer_id
  HAVING COUNT(*) > 1
")

if (count(duplicates) == 0) {
  cat("PASS: customer_id is unique\n")
} else {
  cat("FAIL: duplicate customer_id values found\n")
  showDF(duplicates, truncate = FALSE)
}

File: python/sparklabs/batch/sales/sql/dq_data_quality.py
from sparklabs.batch.sales.sql import utils


def run() -> None:
    duplicates = utils.sql(
        """
        SELECT customer_id, COUNT(*) AS duplicate_count
        FROM customers
        GROUP BY customer_id
        HAVING COUNT(*) > 1
        """
    )
    if duplicates.count() == 0:
        print("PASS: customer_id is unique")
    else:
        print("FAIL: duplicate customer_id values found")
        duplicates.show(truncate=False)


if __name__ == "__main__":
    run()
